lib/generators/gql/gql_generator_base.rb

```diff
@@ -2,40 +2,63 @@
 
 module Gql
   module GqlGeneratorBase
-    extend ActiveSupport::Concern
+    protected
 
-    included do
-      protected
+    # Generate a namedspaced class name with the mutation prefix
+    def prefixed_class_name(prefix)
+      (class_path + ["#{prefix}_#{file_name}"]).map!(&:camelize).join("::")
+    end
 
-      # Generate a namedspaced class name with the mutation prefix
-      def prefixed_class_name(prefix)
-        (class_path + ["#{prefix}_#{file_name}"]).map!(&:camelize).join("::")
-      end
+    def type_map
+      {
+        integer: 'Int',
+        string: 'String',
+        boolean: 'Boolean',
+        decimal: 'Float',
+        datetime: 'GraphQL::Types::ISO8601DateTime',
+        date: 'GraphQL::Types::ISO8601Date',
+        hstore: 'GraphQL::Types::JSON',
+        text: 'String',
+        json: 'GraphQL::Types::JSON'
+      }
+    end
 
-      def type_map
-        {
-          integer: 'Int',
-          string: 'String',
-          boolean: 'Boolean',
-          decimal: 'Float',
-          datetime: 'GraphQL::Types::ISO8601DateTime',
-          date: 'GraphQL::Types::ISO8601Date',
-          hstore: 'GraphQL::Types::JSON',
-          text: 'String',
-          json: 'GraphQL::Types::JSON'
-        }
-      end
-  
-      def map_model_types(model_name)
-        klass = model_name.constantize
-        associations = klass.reflect_on_all_associations(:belongs_to)
-        bt_columns = associations.map(&:foreign_key)
-  
-        klass.columns
-          .reject { |col| bt_columns.include?(col.name) }
-          .reject { |col| type_map[col.type].nil? }
-          .map { |col| {name: col.name, gql_type: type_map[col.type]} }
+    def map_model_types(model_name)
+      klass = model_name.constantize
+      associations = klass.reflect_on_all_associations(:belongs_to)
+      bt_columns = associations.map(&:foreign_key)
+
+      klass.columns
+        .reject { |col| bt_columns.include?(col.name) }
+        .reject { |col| type_map[col.type].nil? }
+        .map { |col| {name: col.name, gql_type: type_map[col.type]} }
+    end
+
+    def root_directory(namespace)
+      "app/graphql/#{namespace.underscore}"
+    end
+
+    def wrap_in_namespace(namespace)
+      namespace = namespace.split('::')
+      namespace.shift if namespace[0].empty?
+
+      code = namespace.each_with_index.map { |name, i| "  " * i + "module #{name}" }.join("\n")
+      code << "\n" << yield(namespace.size) << "\n"
+      code << namespace.size.downto(0).map { |i| "  " * i  + "end" }.join("\n")
+      code
+    end
+
+    def class_with_fields(namespace, name, superclass, fields)
+      wrap_in_namespace(namespace) do |indent|
+        klass = []
+        klass << sprintf("%sclass %s < %s", "  " * indent, name, superclass)
+
+        fields.each do |field|
+          klass << sprintf("%sfield :%s, %s, null: true", "  " * (indent + 1), field[:name], field[:gql_type])
```

Comment:
`:null` should be based on DB default.

Reply:
@sshaw was this comment a future "todo"? I agree with your comment but it looks like this sets null to true all the time (which I'm fine with).

Reply:
Yes, it is I can do it in final commit(s) unless you prefer separate PR.

Reply:
Great, this PR is good.